File: pkg/java/src/main/java/dev/openfga/language/errors/ValidationMetadata.java
package dev.openfga.language.errors;

public class ValidationMetadata extends Metadata {
    private ValidationError errorType;
    private String relation;
    private String typeName;
    private String conditionName;

    // Needed for Jackson deserialization
    public ValidationMetadata() {}

    public ValidationMetadata(String symbol, ValidationError errorType) {
        this(symbol, errorType, null, null, null);
    }

    public ValidationMetadata(
            String symbol, ValidationError errorType, String relation, String typeName, String conditionName) {
        super(symbol);
        this.errorType = errorType;
        this.relation = relation;
        this.typeName = typeName;
        this.conditionName = conditionName;
    }

    public ValidationError getErrorType() {
        return errorType;
    }

    public void setErrorType(ValidationError errorType) {
        this.errorType = errorType;
    }

    public String getRelation() {
        return relation;
    }

    public void setRelation(String relation) {
        this.relation = relation;
    }

    public String getTypeName() {
        return typeName;
    }

    public void setTypeName(String typeName) {
        this.typeName = typeName;
    }

    public String getConditionName() {
        return conditionName;
    }

    public void setConditionName(String conditionName) {
        this.conditionName = conditionName;
    }
}

File: pkg/csharp/src/OpenFga.Language/Errors/ValidationMetadata.cs
using System.Text.Json.Serialization;
using YamlDotNet.Serialization;

namespace OpenFga.Language.Errors;

public class ValidationMetadata
{
    [JsonPropertyName("symbol")]
    [YamlMember(Alias = "symbol")]
    public string Symbol { get; set; } = string.Empty;

    [JsonPropertyName("errorType")]
    [YamlMember(Alias = "errorType")]
    [SharpYaml.Serialization.YamlMember("errorType")]
    public ValidationError ErrorType { get; set; }

    [JsonPropertyName("relation")]
    [YamlMember(Alias = "relation")]
    public string? Relation { get; set; }

    [JsonPropertyName("typeName")]
    [YamlMember(Alias = "typeName")]
    public string? TypeName { get; set; }

    [JsonPropertyName("condition")]
    [YamlMember(Alias = "condition")]
    public string? Condition { get; set; }

    public ValidationMetadata() { }

    public ValidationMetadata(string symbol, ValidationError error)
    {
        Symbol = symbol;
        ErrorType = error;
    }

    public ValidationMetadata(string symbol, ValidationError error, string? relation, string? typeName, string? condition)
    {
        Symbol = symbol;
        ErrorType = error;
        Relation = relation;
        TypeName = typeName;
        Condition = condition;
    }
}
